compiler/il/OMRNode.cpp

```diff
@@ -3391,21 +3391,10 @@ TR_OpaqueMethodBlock*
 OMR::Node::getOwningMethod(TR::Compilation *comp, TR_ByteCodeInfo &bcInfo)
    {
    TR_OpaqueMethodBlock *method = NULL; 
-  
-   if (comp->compileRelocatableCode()) 
-      { 
-      if (0 <= bcInfo.getCallerIndex()) 
-         method = (TR_OpaqueMethodBlock *)(((TR_AOTMethodInfo *)comp->getInlinedCallSite(bcInfo.getCallerIndex())._vmMethodInfo)->resolvedMethod->getPersistentIdentifier()); 
-      else 
-         method = (TR_OpaqueMethodBlock *)(comp->getCurrentMethod()->getPersistentIdentifier()); 
-      } 
-   else  
-      { 
-      if (0 <= bcInfo.getCallerIndex()) 
-         method = (TR_OpaqueMethodBlock *)(comp->getInlinedCallSite(bcInfo.getCallerIndex())._vmMethodInfo); 
-      else 
-         method = (TR_OpaqueMethodBlock *)(comp->getCurrentMethod()->getNonPersistentIdentifier()); 
-      } 
+   if (0 <= bcInfo.getCallerIndex()) 
+      method = comp->getInlinedCallSite(bcInfo.getCallerIndex())._methodInfo; 
+   else 
+      method = comp->getCurrentMethod()->getPersistentIdentifier(); 
```

Comment:
Not really related to this PR but I've always found this frustrating. This `if` / `else` gets repeated in so many places. Would it be possible to modify `getInlinedCallSite` to check for -1 and return the current method instead? Then all of these `if` / `else` can get simplified.

@liqunl thoughts on this?

Reply:
Yeah, but not for `getInlinedCallSite` as it returns `TR_InlinedCallSite` which the outermost method doesn't have. But we can add a set of Compilation APIs that takes any caller index and return opaque method block, resolved method or resolved method symbol.

Reply:
For resolved method and method symbol, we can just modify `OMR::Compilation::getInlinedResolvedMethod(Symbol)` to check for `-1` and use the current method in that case. We could add a similar API to get the opaque method block.